Standard_Functions_RIbbon.py

```diff
@@ -0,0 +1,380 @@
+            True,
+        )
+        if reply[1]:
+            # user clicked OK
+            replyText = reply[0]
+        else:
+            # user clicked Cancel
+            replyText = reply[0]  # which will be "" if they clicked Cancel
+        return str(replyText)
+
+
+def RestartDialog(message="", includeIcons=False):
+    """_summary_
+        shows a restart dialog
+    Returns:
+        string: returns 'yes' if restart now is clicked.
+        otherwise returns 'no'
+    """
+    from PySide.QtWidgets import QMessageBox
+
+    # Save the preferences before restarting
+    App.saveParameter()
+
+    # Set the message
+    if message == "":
+        message = "You must restart FreeCAD for changes to take effect."
+
+    # Set the messagebox
+    msgBox = QMessageBox()
+    msgBox.setIcon(QMessageBox.Warning)
+    msgBox.setText(message)
+    msgBox.setWindowTitle("FreeCAD Ribbon")
+    # Set the buttons and default button
+    msgBox.setStandardButtons(QMessageBox.Yes | QMessageBox.No)
+    msgBox.setDefaultButton(QMessageBox.No)
+    msgBox.button(QMessageBox.Yes).setText("Restart now")
+    msgBox.button(QMessageBox.No).setText("Restart later")
+    if includeIcons is True:
+        msgBox.button(QMessageBox.No).setIcon(Gui.getIcon("edit_Cancel.svg"))
+        msgBox.button(QMessageBox.Yes).setIcon(Gui.getIcon("edit_OK.svg"))
+
+    reply = msgBox.exec_()
+    if reply == QMessageBox.Yes:
+        return "yes"
+    if reply == QMessageBox.No:
+        return "no"
+
+
+def restart_freecad():
+    from PySide import QtWidgets, QtCore
+
+    """Shuts down and restarts FreeCAD"""
+
+    args = QtWidgets.QApplication.arguments()[1:]
+    if Gui.getMainWindow().close():
+        QtCore.QProcess.startDetached(
+            QtWidgets.QApplication.applicationFilePath(), args
+        )
+
+    return
+
+
+def SaveDialog(files, OverWrite: bool = True):
+    """
+    files must be like:\n
+    files = [\n
+        ('All Files', '*.*'),\n
+        ('Python Files', '*.py'),\n
+        ('Text Document', '*.txt')\n
+    ]\n
+    \n
+    OverWrite:\n
+    If True, file will be overwritten\n
+    If False, only the path+filename will be returned\n
+    """
+    import tkinter as tk
+    from tkinter.filedialog import asksaveasfile
+    from tkinter.filedialog import askopenfilename
+
+    # Create the window
+    root = tk.Tk()
+    # Hide the window
+    root.withdraw()
+
+    if OverWrite is True:
+        file = asksaveasfile(filetypes=files, defaultextension=files)
+        if file is not None:
+            return file.name
+    if OverWrite is False:
+        file = askopenfilename(filetypes=files, defaultextension=files)
+        if file is not None:
+            return file
+
+
+def GetLetterFromNumber(number: int, UCase: bool = True):
+    # from openpyxl.utils import get_column_letter
+
+    # Letter = get_column_letter(number)
+
+    # # If UCase is true, convert to upper case
+    # if UCase is True:
+    #     Letter = Letter.upper()
```

Comment:
Why is this as a comment? If it is not needed, please remove it or leave a comment, why it is there as a comment.

Reply:
I've a standard functions repo where this is a copy of. This is probably a left over from my other wb's. it is removed.